moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -896,29 +896,49 @@ bool TrajectoryExecutionManager::validate(const TrajectoryExecutionContext &cont
     ROS_WARN_NAMED("traj_execution", "Failed to validate trajectory: couldn't receive full current joint state within 1s");
     return false;
   }
+  current_state->enforceBounds();
```

Comment:
Could you add a comment why this is necessary? Shouldn't the current state already be within bounds?